### PR TITLE
feat(github-actions): support ratchet comments

### PR DESCRIPTION
**Pull request by art-shutter**
## Changes

This adds support for comments format used by [`ratchet` tool](https://github.com/sethvargo/ratchet).

## Context

The `ratchet` tool is a locally run tool that helps update github workflows dependencies locally. This is especially useful in monorepos containing several separately-versioned github actions / workflows. As a `ratchet` user I would like to keep the functionality ratchet is offering me and combine it with `renovate`. This PR makes sure the desired pin is extracted from `ratchet` comments and that `ratchet` comments don't get deleted by renovate, including the `ratchet:exclude` comments.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/github-actions/readme.md
Comment (art-shutter) [reply lib/modules/manager/github-actions/readme.md]: I mean the ratchet comments with pinned versions would be taken into consideration as well as `# pin @2.1.0` comments. This also means that ratchet comments don't get deleted/modified after being parsed.
It is also the case, that `# ratchet:exclude` comments would not be deleted.

File: lib/modules/manager/github-actions/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/github-actions/readme.md:44]:
```suggestion
The `github-action` manager understands `ratchet` comments, like `# ratchet:actions/checkout@v2.1.0`.
This means that Renovate will:

- update the version of a _pinned_ Ratchet version if needed
- not delete/modify Ratchet comments after parsing them
- keep `# ratchet:exclude` comments

```

> I mean the ratchet comments with pinned versions would be taken into consideration as well as `# pin @2.1.0` comments. This also means that ratchet comments don't get deleted/modified after being parsed. It is also the case, that `# ratchet:exclude` comments would not be deleted.

I think a short list of the supported usecases/behavior would help understand what's going to happen.

Please review my suggestion for accuracy before applying it. :wink:

**Review by HonkingGoose — APPROVED** [review]
I'm happy with the docs, I'll let the maintainers do the code review.

**Review by viceice — APPROVED** [review]

File: lib/modules/manager/github-actions/extract.ts
Comment (HonkingGoose) [inline lib/modules/manager/github-actions/extract.ts:18]: @TWiStErRob do you want to check this one as well?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Maybe we should let @TWiStErRob review the regex too.

File: lib/modules/manager/github-actions/extract.ts
Comment (TWiStErRob) [reply lib/modules/manager/github-actions/extract.ts:18]: No. This one needs a lot of unit tests. 🧐

**Review by HonkingGoose — APPROVED** [review]

File: lib/modules/manager/github-actions/readme.md
Comment (HonkingGoose) [reply lib/modules/manager/github-actions/readme.md:48]: @art-shutter and @rarkins can you check this review comment?

Comment (viceice) [reply lib/modules/manager/github-actions/readme.md:48]: @mkurde please open a new discussion and reference this pr

**Pull request merged**
